### ISSUE TITLE
Detect multi-initialization of same table key

### ISSUE THREAD

**Issue by bjorn** [mpeterv/luacheck#48]
Just a suggestion I thought would be useful to check for:

```lua
local table = {
    key = "foo",
    key = "bar",
    ["key"] = "baz",  -- a little tricker to check I guess
}
```

Would be nice if `luacheck` would warn about this case, since I don't see it ever being useful, but Lua will not complain and just have `table.key == "baz"`.

**Comment by mpeterv** [comment]
Implemented on master branch, looks like this;

```
Checking test.lua                                 2 warnings

    test.lua:2:5: unused field key
    test.lua:3:5: unused field key

Total: 2 warnings / 0 errors in 1 file
```

**Comment by mpeterv** [comment]
Changed to

```
Checking test.lua                                 2 warnings

    test.lua:2:5: value assigned to field 'key' is unused
    test.lua:3:5: value assigned to field 'key' is unused

Total: 2 warnings / 0 errors in 1 file
```

in line with `unused value` warnings for local variables. It also works for non-string keys, including mixed array/hash tables:

```
local table = {1, [1] = 2}
```

```
Checking test.lua                                 1 warning

    test.lua:1:16: value assigned to index 1 is unused

Total: 1 warning / 0 errors in 1 file
```

**Comment by bjorn** [comment]
That was fast! Looking forward to the next luacheck release. :-)
